Automattic/wp-calypso
# BlazePress search campaign / post endpoint

Related to #

## Proposed Changes

* Added new blaze/posts paginated endpoint
* Applied infinite scrolling with server pagination
* Wired the search bar and filters in both campaigns and posts list
* revamped the main page and reduced the code to include the new GET endpoint
* Added missing styles and Skeleton loader for entries

## Testing Instructions
- Add this diff in your wpcom sandbox D109592
- checkout this branch (https://github.tumblr.net/Tumblr/a8c-dsp/pull/91) in the DSP server 
- Go to /advertising page in calypso
- You should see all the possible posts to be blazed from the current selected blog. If the amount is big enough (20 items) it will add more entries as you scroll. You can change this number at the wpcom diff
- You should be able to search by post title and order by different options in the search bar
- Go to campaigns tab. You should see all the user campaigns
- You should be able to search campaigns and filter them by status

## Pre-merge Checklist


![image](https://github.com/Automattic/wp-calypso/assets/43957544/6de90b52-9a7b-4b98-acd8-134369c48972)

You won't see images in the campaign list since there's no image serving in your locale system when running the DSP server


![image](https://github.com/Automattic/wp-calypso/assets/43957544/9f59f399-9161-4072-aad7-5b1e6c59bae4)

There's some nice algorithm added at the wpcom side to get the first canonical image from a post. It works quite good!
